plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx

```diff
@@ -0,0 +1,128 @@
+    // states
+    const [isLoading, setIsLoading] = useState(false);
+    const [indexPatternSelected, setIndexPatternSelected] = useState<IIndexPattern>();
+    const [error, setError] = useState();
+
+    useEffect(() => {
+        setIsLoading(true);
+        initSearchBar();
+        setIsLoading(false);
+    }, []);
+
+    /**
+     * Initialize the searchbar props with the corresponding index pattern and filters
+     */
+    const initSearchBar = async () => {
+        const indexPattern = await getIndexPattern(props?.defaultIndexPatternID);
+        setIndexPatternSelected(indexPattern);
+        const filters = await getInitialFilters(indexPattern);
+        filterManager.setFilters(filters);
+    }
+
+    /**
+     * Return the index pattern data by ID.
+     * If not receive a ID return the default index from the index pattern service
+     * @returns 
+     */
+    const getIndexPattern = async (indexPatternID?: string) => {
+        const indexPatternService = getDataPlugin().indexPatterns as IndexPatternsContract;
+        if(indexPatternID){
+            try{
+                return await indexPatternService.get(indexPatternID);
+            }catch(error){
+                // when the index pattern id not exists will get the default
+                console.error(error);
+                return await indexPatternService.getDefault();
+            }
+        }else{
+            return await indexPatternService.getDefault();
+        }
+       
+    }
+
+    /**
+     * Return the initial filters considering if hook receives initial filters
+     * When the default index pattern is the same like the received preserve the filters
+     * @param indexPattern 
+     * @returns 
+     */
+    const getInitialFilters = async (indexPattern: IIndexPattern) => {
+        const indexPatternService = getDataPlugin().indexPatterns as IndexPatternsContract;
+        let initialFilters: Filter[] = [];
+        if(props?.filters){
+            return props?.filters;
+        }
+        if(indexPattern){
+            // get filtermanager and filters
+            // if the index is the same, get filters stored
+            // else clear filters
+            const defaultIndexPattern = await indexPatternService.getDefault() as IIndexPattern;
+            initialFilters = defaultIndexPattern.id === indexPattern.id ? filterManager.getFilters(): []
+        }else{
+            initialFilters = [];
+        }
+        return initialFilters.filter(filter => filter.meta.controlledBy !== AUTHORIZED_AGENTS);
+    }
+
+    /**
+     * Search bar properties necessary to render and initialize the osd search bar component
+     */
+    const searchBarProps: Partial<SearchBarProps> = {
+        isLoading,
+        ...indexPatternSelected && { indexPatterns: [indexPatternSelected] }, // indexPattern cannot be empty or empty []
+        filters: filterManager.getFilters(),
+        query: query,
+        timeHistory: timeHistory,
```

Comment:
```suggestion
        timeHistory,
```

Reply:
Idem

Reply:
Resolved in https://github.com/wazuh/wazuh-kibana-app/pull/5960/commits/2348731fb1410f74abdaeee6920585942d2e2c0a